pkgs/by-name/cl/cloudquery/package.nix

```diff
@@ -0,0 +1,52 @@
+{
+  lib,
+  buildGoModule,
+  fetchFromGitHub,
+}:
+
+buildGoModule rec {
+  pname = "cloudquery";
+  version = "6.11.2";
+
+  src = fetchFromGitHub {
+    owner = "cloudquery";
+    repo = "cloudquery";
+    rev = "refs/tags/cli-v${version}";
+    hash = "sha256-zVzHPAN/UAVQF56qcoIFOqtX/Aek3+1lO2O965UYxPM=";
+  };
+
+  vendorHash = "sha256-9YvsVflHr4QSVCYCcWg1Okmncc1sNteMJEAdz92BIgI=";
+
+  CGO_ENABLED = 0;
+
+  modRoot = "cli";
+
+  # Some checks fail to download deps
+  doCheck = false;
+
+  ldflags = [
+    "-s"
+    "-w"
+    "-X github.com/cloudquery/cloudquery/cli/cmd.Version=${version}"
+  ];
+
+  postInstall = ''
+    mv $out/bin/cli $out/bin/cloudquery
+    rm $out/bin/gen
+  '';
+
```

Comment:
```suggestion

  doInstallCheck = true;

  nativeInstallCheckInputs = [
    versionCheckHook
  ];
```

Reply:
With those lines above, `nixpkgs-review` threw the following error:

`error: undefined variable 'versionCheckHook'`

Reply:
> `error: undefined variable 'versionCheckHook'`

You have to add it to argument of this function of package definition. I don't need to make sure my diff can be run directly after applying. That's not how suggestions work.

Reply:
> With those lines above, `nixpkgs-review` threw the following error:
> 
> `error: undefined variable 'versionCheckHook'`

You also need to import that and put it in `nativeBuildInputs`.

<details><summary>Full diff (click to expand)</summary>

```diff
diff --git a/pkgs/by-name/cl/cloudquery/package.nix b/pkgs/by-name/cl/cloudquery/package.nix
index 3b245cf82414..c7e0f150e849 100644
--- a/pkgs/by-name/cl/cloudquery/package.nix
+++ b/pkgs/by-name/cl/cloudquery/package.nix
@@ -2,6 +2,7 @@
   lib,
   buildGoModule,
   fetchFromGitHub,
+  versionCheckHook,
 }:
 
 buildGoModule rec {
@@ -21,6 +22,10 @@ buildGoModule rec {
 
   modRoot = "cli";
 
+  nativeBuildInputs = [
+    versionCheckHook
+  ];
+
   # Some checks fail to download deps
   doCheck = false;
 
@@ -35,6 +40,11 @@ buildGoModule rec {
     rm $out/bin/gen
   '';
 
+  doInstallCheck = true;
+  nativeInstallCheckInputs = [
+    versionCheckHook
+  ];
+
   meta = {
     description = "Data movement tool to sync data from any source to any destination";
     mainProgram = "cloudquery";
```

See the docs for more information: https://github.com/NixOS/nixpkgs/blob/87c4f4d7ba9e21ff7decf22d6d3a11f8e5cf3b60/doc/hooks/versionCheckHook.section.md

Reply:
The checkPhase section still throws errors, here's the build log with the errors:
[cloudquery-x86_64-linux.log](https://github.com/user-attachments/files/18033612/cloudquery-x86_64-linux.log)

What do you recommend here? Thanks!

Reply:
@mrgiles Those errors are coming from upstream's tests (`doCheck = false` to disable them).

Our version check (with `doInstallCheck = true` and `versionCheckHook` in `nativeInstallCheckInputs`) works.

<details><summary>Build log (with `doCheck = false`)</summary>

```
Using versionCheckHook
Running phase: unpackPhase
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking source archive /nix/store/fiv59ma73fh8n09czqkzxlkhqk4a0iw0-source
source root is source
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "patchPhase" }
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Running phase: configurePhase
@nix { "action": "setPhase", "phase": "configurePhase" }
Running phase: buildPhase
@nix { "action": "setPhase", "phase": "buildPhase" }
Building subPackage .
Building subPackage ./cmd
Building subPackage ./internal/analytics
Building subPackage ./internal/api
Building subPackage ./internal/auth
Building subPackage ./internal/docs
Building subPackage ./internal/enum
Building subPackage ./internal/env
Building subPackage ./internal/hub
Building subPackage ./internal/otel
Building subPackage ./internal/publish
Building subPackage ./internal/publish/images
Building subPackage ./internal/specs/v0
Building subPackage ./internal/specs/v0/gen
Building subPackage ./internal/tablenamechanger
Building subPackage ./internal/team
Building subPackage ./internal/transformer
Building subPackage ./internal/transformerpipeline
Building subPackage ./internal/uuid
buildPhase completed in 51 seconds
Running phase: installPhase
@nix { "action": "setPhase", "phase": "installPhase" }
Running phase: fixupPhase
@nix { "action": "setPhase", "phase": "fixupPhase" }
shrinking RPATHs of ELF executables and libraries in /nix/store/4vyjdwpcqkf7bi6v22vfyzgdrymm3f74-cloudquery-6.11.2
shrinking /nix/store/4vyjdwpcqkf7bi6v22vfyzgdrymm3f74-cloudquery-6.11.2/bin/cloudquery
patchelf: cannot find section '.dynamic'. The input file is most likely statically linked
checking for references to /build/ in /nix/store/4vyjdwpcqkf7bi6v22vfyzgdrymm3f74-cloudquery-6.11.2...
patchelf: cannot find section '.dynamic'. The input file is most likely statically linked
patching script interpreter paths in /nix/store/4vyjdwpcqkf7bi6v22vfyzgdrymm3f74-cloudquery-6.11.2
stripping (with command strip and flags -S -p) in  /nix/store/4vyjdwpcqkf7bi6v22vfyzgdrymm3f74-cloudquery-6.11.2/bin
Running phase: installCheckPhase
@nix { "action": "setPhase", "phase": "installCheckPhase" }
Executing versionCheckPhase
Did not find version 6.11.2 in the output of the command /nix/store/4vyjdwpcqkf7bi6v22vfyzgdrymm3f74-cloudquery-6.11.2/bin/cloudquery --help
CloudQuery CLI

High performance data integration at scale.

Find more information at:
	https://www.cloudquery.io

Usage:
  cloudquery [command]

Available Commands:
  addon           Addon commands
  init            Generate a configuration file for a sync
  login           Login to CloudQuery Hub.
  logout          Log out of CloudQuery Hub.
  migrate         Update schema of your destinations based on the latest changes in sources from your configuration
  plugin          Plugin commands
  switch          Switches between teams.
  sync            Sync resources from configured source plugins to destinations
  tables          Generate documentation for all supported tables of source plugins specified in the spec(s)
  test-connection Test plugin connections to sources and/or destinations
  validate-config Validate config

Flags:
      --cq-dir string            directory to store cloudquery files, such as downloaded plugins (default ".cq")
  -h, --help                     help for cloudquery
      --invocation-id uuid       useful for when using Open Telemetry integration for tracing and logging to be able to correlate logs and traces through many services (default <NEW-RANDOM-UUID>)
      --log-console              enable console logging
      --log-file-name string     Log filename (default "cloudquery.log")
      --log-format string        Logging format (json, text) (default "text")
      --log-level string         Logging level (trace, debug, info, warn, error) (default "info")
      --no-log-file              Disable logging to file
      --telemetry-level string   Telemetry level (none, errors, stats, all) (default "all")
  -v, --version                  version for cloudquery

Use "cloudquery [command] --help" for more information about a command.
Successfully managed to find version 6.11.2 in the output of the command /nix/store/4vyjdwpcqkf7bi6v22vfyzgdrymm3f74-cloudquery-6.11.2/bin/cloudquery --version
cloudquery version 6.11.2
Finished versionCheckPhase
no Makefile or custom installCheckPhase, doing nothing
```
</details>